### ISSUE TITLE
EMR instance gets "forces new instance" when no changes made

### ISSUE THREAD

**Issue by synhershko** [terraform-providers/terraform-provider-aws#5075]
Code to launch the EMR cluster:

```
resource "aws_emr_cluster" "parqueting" {
  name          = "Parqueting"
  release_label = "emr-5.14.0"
  applications  = ["Spark", "Ganglia"]

  log_uri = "s3n://my-emr-logs/parqueting/"

  ec2_attributes {
    subnet_id                         = "${var.subnet_id}"
    instance_profile                  = "${aws_iam_instance_profile.emr_profile.arn}"
    key_name                          = "${var.keyname}"
  }

  configurations = "emr-parqueting-configurations.json"

  master_instance_type  = "m4.large"
  core_instance_type    = "c4.2xlarge"
  core_instance_count   = 1

  tags {
    name     = "parqueting"
  }
  service_role = "${aws_iam_role.iam_emr_service_role.arn}"
}
```

After creation, whenever I try to change other things the EMR cluster is planned for recreation as well although nothing changed there:

```
      ec2_attributes.0.emr_managed_master_security_group: "sg-057fe7bb416c5983f" => "" (forces new resource)
      ec2_attributes.0.emr_managed_slave_security_group:  "sg-01ac84737e433b8e1" => "" (forces new resource)
      ec2_attributes.0.instance_profile:                  "arn:aws:iam::744522205193:instance-profile/emr_profile" => "arn:aws:iam::744522205193:instance-profile/emr_profile"
      ec2_attributes.0.key_name:                          "sa-staging-emr" => "sa-staging-emr"
      ec2_attributes.0.service_access_security_group:     "sg-02cc532fcc7432a57" => "" (forces new resource)
```

If I fill in the `ec2_attributes` with those 3 "missing" attributes everything works correctly and the terraform doesn't attempt to recreate the cluster

**Comment by blckct** [comment]
TF doesn't seem to expect those attributes will be filled-in by default, should probably be marker as computed. It doesn't complain about `service_access_security_group` for me though, just the other two.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.45.0 of the Terraform AWS Provider, Thursday next week. Thanks to @blckct for the implementation. 👍
